### PR TITLE
Laboratory work #1, Babiy Alexander - 19FPL1

### PR DESCRIPTION
**Pull request by asbabiy**

### CODE REVIEW

**Comment by ValeriyaKuznetsova** [comment]
Unfortunately, the deadline is not that far. Please, push changes so that we can review them.

File: lab_1/main.py [hunk 1]
```diff
@@ -25,7 +33,7 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
+    return [x for x in tokens if x not in set(stop_words)] if type(tokens) is list else []
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: consider removing "if type()" statements from here, the line is hard to read

File: lab_1/main.py [hunk 2]
```diff
@@ -36,7 +44,7 @@ def calculate_frequencies(tokens: list) -> dict:
     e.g. tokens = ['weather', 'sunny', 'man', 'happy']
     --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
     """
-    pass
+    return {token: tokens.count(token) for token in tokens} if (type(tokens) is list and all(tokens)) else {}
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: same about "if type"

File: lab_1/main.py [hunk 3]
```diff
@@ -69,7 +81,32 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
+
+    if all(eval(x) for x in checks):
+        context = []
+        word_indices = [idx for idx, el in enumerate(tokens) if el == word]
+        window = (left_context_size, right_context_size)
+
+        for x in word_indices:
+            if window[0] > 0 and window[1] > 0:
+                context.append(tokens[x - window[0]:x + window[1] + 1])
+            elif window[0] < 1 and window[1] > 0:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: do you need "window[0]<1"? if it's > 1, the first 'if' will work

File: lab_1/main.py [hunk 4]
```diff
@@ -69,7 +81,32 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
+                context.append(tokens[x:x + window[1] + 1])
+            elif window[0] > 0 and window[1] < 1:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: do you need "window[1]<1"? if it's > 1, the first 'if' will work

File: lab_1/main.py [hunk 5]
```diff
@@ -69,7 +87,32 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
...
+    checks = [
+        isinstance(tokens, list),
+        isinstance(word, str),
+        isinstance(left_context_size, int),
+        isinstance(right_context_size, int),
+        not isinstance(left_context_size, bool),
+        not isinstance(right_context_size, bool)
+    ]
+
+    if all(checks):
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: You can frame only one line in if statement, and everything else would not be in if.

File: lab_1/main.py [hunk 6]
```diff
@@ -87,16 +130,38 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
...
+    checks = [
+        isinstance(tokens, list),
+        isinstance(word, str),
+        isinstance(left_n, int),
+        isinstance(right_n, int),
+        not isinstance(left_n, bool),
+        not isinstance(right_n, bool)
+    ]
+
+    if all(checks):
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: same about one line and if

File: lab_1/main.py [hunk 7]
```diff
@@ -87,16 +130,38 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
...
+        window = (left_n, right_n)
+        concordance = get_concordance(tokens, word, left_n, right_n)
+
+        if window[0] > 0 and window[1] > 0:
+            return [[token[0], token[-1]] for token in concordance]
+        if window[1] > 0:
+            return [[token[-1]] for token in concordance]
+        if window[0] > 0:
+            return [[token[0]] for token in concordance]
+        return []
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py:145]: do you need two lines with "return []"?

File: lab_1/main.py [hunk 8]
```diff
@@ -105,7 +170,10 @@ def write_to_file(path_to_file: str, content: list):
     """
     Writes the result in a file
     """
-    pass
+
+    with open(path_to_file, 'w', encoding='utf-8') as data:
+        for i in content:
+            data.write(f'{repr(i)}\n')
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: can you suggest an easier way of putting words together?

File: lab_1/main.py [hunk 9]
```diff
@@ -125,4 +193,24 @@ def sort_concordance(tokens: list, word: str, left_context_size: int, right_cont
...
+        isinstance(tokens, list),
+        isinstance(word, str),
+        isinstance(left_context_size, int),
+        isinstance(right_context_size, int),
+        not isinstance(left_context_size, bool),
+        not isinstance(right_context_size, bool),
+        isinstance(left_sort, bool)
+    ]
+
+    if all(checks):
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: same about if

File: lab_1/main.py [hunk 10]
```diff
@@ -125,4 +193,24 @@ def sort_concordance(tokens: list, word: str, left_context_size: int, right_cont
...
+        concordance = get_concordance(tokens, word, left_context_size, right_context_size)
+        window = (left_context_size, right_context_size)
+
+        if left_sort and window[0] > 0:
+            return sorted(concordance)
+        if not left_sort and window[1] > 0:
+            return sorted(concordance, key=lambda x: x[x.index(word) + 1])
+        return []
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py:210]: two lines with "return []"

File: lab_1/report.txt [hunk 11]
```diff
@@ -0,0 +1,2 @@
+['man', 'is', 'happy', 'the', 'dog', 'is']
```
Comment (ValeriyaKuznetsova) [inline lab_1/report.txt:1]: make it like:
man is happy the dog is

File: lab_1/start.py [hunk 12]
```diff
@@ -2,17 +2,38 @@
...
+    sorted_concordance = main.sort_concordance(tokens, 'people', 2, 3, False)
+    print(f'SORT CONCORDANCE >> {sorted_concordance[:2]}')
+
+    main.write_to_file('report.txt', sorted_concordance)
+    print('WRITE TO FILE >> SUCCESSFUL')
+
+    RESULT = sorted_concordance
     # DO NOT REMOVE NEXT LINE - KEEP IT INTENTIONALLY LAST
-    assert RESULT == [(), ()], 'Concordance not working'
+    assert RESULT, 'Concordance not working'
```
Comment (ValeriyaKuznetsova) [inline lab_1/start.py]: after == you should put an expected result, for example, 
assert RESULT == [['man', 'is', 'happy', 'the', 'dog', 'is'], ['dog', 'is', 'happy', 'but', 'the', 'cat']]

File: lab_1/main.py (hunk 8 above)
Comment (ValeriyaKuznetsova) [reply lab_1/main.py]: they should be transformed into text

File: lab_1/main.py (hunk 5 above)
Comment (asbabiy) [reply lab_1/main.py]: What do you mean?

Comment (asbabiy) [reply lab_1/main.py]: Got it

**Comment by ValeriyaKuznetsova** [comment]
@bac03704-byte 
everything seems good!

**Pull request closed without merging**